Unhandled exception in vault_raft_snapshot_agent_config on plan

This issue was originally add to #1354 https://github.com/hashicorp/terraform-provider-vault/issues/1354#issuecomment-1046113607 which is unrelated.

The jist:

Only one object in the config seems to be causing it.  If I remove this object, i do not get the panic:

```hcl
# Configure snapshots of the raft integrated storage
resource "vault_raft_snapshot_agent_config" "default" {
  name             = "s3_daily"
  interval_seconds = 86400
  retain           = var.snapshot_retention_count
  path_prefix      = var.snapshot_bucket_path_prefix
  storage_type     = "aws-s3"

  aws_s3_bucket     = data.aws_s3_bucket.snapshots.bucket
  aws_s3_region     = data.aws_region.current.name
  aws_s3_enable_kms = true
  aws_s3_kms_key    = var.snapshot_kms_key_id
}

data "aws_s3_bucket" "snapshots" {
  bucket = var.snapshot_bucket
}

data "aws_region" "current" {}
```

Similar crash output as above:

```
module.vault_config.vault_raft_snapshot_agent_config.default: Creating...
╷
│ Error: Plugin did not respond
│
│   with module.vault_config.vault_raft_snapshot_agent_config.default,
│   on ../../../../../../../providers/vault/modules/vault-base-config/snapshots.tf line 2, in resource "vault_raft_snapshot_agent_config" "default":
│    2: resource "vault_raft_snapshot_agent_config" "default" {
│
│ The plugin encountered an error, and failed to respond to the plugin.(*GRPCProvider).ApplyResourceChange call. The plugin logs may
│ contain more details.
╵

Stack trace from the terraform-provider-vault_v3.3.0_x4 plugin:

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x30 pc=0x1a8dc5a]

goroutine 85 [running]:
github.com/hashicorp/terraform-provider-vault/vault.readSnapshotAgentConfigResource(0xc0005ac100, {0x1cd5e40, 0xc0005d8280})
	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/hashicorp/terraform-provider-vault/vault/resource_raft_snapshot_agent_config.go:321 +0x31a
github.com/hashicorp/terraform-provider-vault/vault.createOrUpdateSnapshotAgentConfigResource(0x0, {0x1cd5e40, 0xc0005d8280})
	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/hashicorp/terraform-provider-vault/vault/resource_raft_snapshot_agent_config.go:298 +0x287
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).create(0x1b93740, {0x1f55b08, 0xc000724000}, 0x2, {0x1cd5e40, 0xc0005d8280})
	/opt/teamcity-agent/work/5d79fe75d4460a2f/pkg/mod/github.com/hashicorp/terraform-plugin-sdk/v2@v2.10.0/helper/schema/resource.go:329 +0x178
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).Apply(0xc0007f1340, {0x1f55b08, 0xc000724000}, 0xc0004ef790, 0xc000785d00, {0x1cd5e40, 0xc0005d8280})
	/opt/teamcity-agent/work/5d79fe75d4460a2f/pkg/mod/github.com/hashicorp/terraform-plugin-sdk/v2@v2.10.0/helper/schema/resource.go:467 +0x871
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ApplyResourceChange(0xc00000c738, {0x1f55b08, 0xc000724000}, 0xc0000a0c30)
	/opt/teamcity-agent/work/5d79fe75d4460a2f/pkg/mod/github.com/hashicorp/terraform-plugin-sdk/v2@v2.10.0/helper/schema/grpc_provider.go:977 +0xd8a
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ApplyResourceChange(0xc0001d7000, {0x1f55bb0, 0xc0005cc2a0}, 0xc00033bb90)
	/opt/teamcity-agent/work/5d79fe75d4460a2f/pkg/mod/github.com/hashicorp/terraform-plugin-go@v0.5.0/tfprotov5/tf5server/server.go:603 +0x30e
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ApplyResourceChange_Handler({0x1c9fa00, 0xc0001d7000}, {0x1f55bb0, 0xc0005cc2a0}, 0xc00091d500, 0x0)
	/opt/teamcity-agent/work/5d79fe75d4460a2f/pkg/mod/github.com/hashicorp/terraform-plugin-go@v0.5.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:380 +0x170
google.golang.org/grpc.(*Server).processUnaryRPC(0xc0006ca8c0, {0x1f6be60, 0xc00018f080}, 0xc0001537a0, 0xc00081c390, 0x27373e0, 0x0)
	/opt/teamcity-agent/work/5d79fe75d4460a2f/pkg/mod/google.golang.org/grpc@v1.41.0/server.go:1279 +0xccf
google.golang.org/grpc.(*Server).handleStream(0xc0006ca8c0, {0x1f6be60, 0xc00018f080}, 0xc0001537a0, 0x0)
	/opt/teamcity-agent/work/5d79fe75d4460a2f/pkg/mod/google.golang.org/grpc@v1.41.0/server.go:1608 +0xa2a
google.golang.org/grpc.(*Server).serveStreams.func1.2()
	/opt/teamcity-agent/work/5d79fe75d4460a2f/pkg/mod/google.golang.org/grpc@v1.41.0/server.go:923 +0x98
created by google.golang.org/grpc.(*Server).serveStreams.func1
	/opt/teamcity-agent/work/5d79fe75d4460a2f/pkg/mod/google.golang.org/grpc@v1.41.0/server.go:921 +0x294

Error: The terraform-provider-vault_v3.3.0_x4 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -2,6 +2,7 @@
 BUGS:
 * `resource/identity_group`: Report an error upon duplicate resource creation failure. Document group name caveats. ([#1352](https://github.com/hashicorp/terraform-provider-vault/pull/1352))
 * `resource/pki_secret_backend_root_sign_intermediate`: Fix panic when reading `ca_chain` from Vault ([#1357](https://github.com/hashicorp/terraform-provider-vault/issues/1357)
+* `resource/raft_snapshot_agent_config`: Properly handle nil response on read ([#1360](https://github.com/hashicorp/terraform-provider-vault/pull/1360))
 
 ## 3.3.0 (February 17, 2022)
 FEATURES:

--- a/vault/resource_raft_snapshot_agent_config.go
+++ b/vault/resource_raft_snapshot_agent_config.go
@@ -11,8 +11,10 @@ import (
 	"github.com/hashicorp/vault/api"
 )
 
-var snapshotAutoPath = "sys/storage/raft/snapshot-auto/config/%s"
-var allowedStorageTypes = []string{"local", "azure-blob", "aws-s3", "google-gcs"}
+var (
+	snapshotAutoPath    = "sys/storage/raft/snapshot-auto/config/%s"
+	allowedStorageTypes = []string{"local", "azure-blob", "aws-s3", "google-gcs"}
+)
 
 func raftSnapshotAgentConfigResource() *schema.Resource {
 	fields := map[string]*schema.Schema{
@@ -196,7 +198,6 @@ func buildConfigFromResourceData(d *schema.ResourceData) (map[string]interface{}
 		} else {
 			return nil, errors.New("specified local storage without setting local_max_space")
 		}
-
 	}
 
 	if storageType == "aws-s3" {
@@ -306,11 +307,12 @@ func readSnapshotAgentConfigResource(d *schema.ResourceData, meta interface{}) e
 	log.Printf("[DEBUG] Reading %q", configPath)
 
 	resp, err := client.Logical().Read(configPath)
-	if err != nil && util.Is404(err) {
+	if resp == nil || (err != nil && util.Is404(err)) {
 		log.Printf("[WARN] %q not found, removing from state", name)
 		d.SetId("")
 		return nil
-	} else if err != nil {
+	}
+	if err != nil {
 		return fmt.Errorf("error reading %q: %s", configPath, err)
 	}
 

